js/main.js

```diff
@@ -11,49 +11,27 @@ $avatarUrl.addEventListener('input', function (event) {
   $imageSelect.setAttribute('src', event.target.value);
 });
 
+function displaySubmitErrorModal() {
+  document.querySelector('.container').classList.add('modal-check');
+  document.querySelector('#modal').classList.remove('hidden');
+}
+
+var $allInputs = document.querySelectorAll('input');
 $formSelect.addEventListener('submit', function (event) {
   event.preventDefault();
-  if (!$userNameInfo.value.trim()) { // prevents user from having username with spaces
-    document.querySelector('.container').classList.add('modal-check');
-    document.querySelector('#modal').classList.remove('hidden');
-    return;
-  } else {
-    data.profile.username = $userNameInfo.value;
-  }
-  if (!$avatarUrl.value.trim()) { // prevents user from having no images
-    document.querySelector('.container').classList.add('modal-check');
-    document.querySelector('#modal').classList.remove('hidden');
-    return;
-  } else {
-    data.profile.avatarUrl = $avatarUrl.value;
-  }
-
-  if (!$userFullNameInfo.value.trim()) { // prevents user from having name with all spaces
-    document.querySelector('.container').classList.add('modal-check');
-    document.querySelector('#modal').classList.remove('hidden');
-    return;
-  } else {
-    data.profile.fullName = $userFullNameInfo.value;
-  }
-
-  if (!$userLocationInfo.value.trim()) { // prevents user from having location with  all spaces
-    document.querySelector('.container').classList.add('modal-check');
-    document.querySelector('#modal').classList.remove('hidden');
-    return;
-  } else {
-    data.profile.location = $userLocationInfo.value;
+  for (var i = 0; i < $allInputs.length; i++) {
+    if (!$allInputs[i].value.trim()) {
+      displaySubmitErrorModal();
+      swapView('edit-profile');
```

Comment:
If this runs, the user should already be on the edit profile view, so this shouldn't be necessary 👍